image/tarexport/tarexport.go

```diff
@@ -15,10 +16,11 @@ const (
 )
 
 type manifestItem struct {
-	Config   string
-	RepoTags []string
-	Layers   []string
-	Parent   image.ID `json:",omitempty"`
+	Config       string
+	RepoTags     []string
+	Layers       []string
+	Parent       image.ID                                  `json:",omitempty"`
+	LayerSources map[layer.DiffID]*distribution.Descriptor `json:",omitempty"`
```

Comment:
it's unfortunate this map is indexed by `layer.DiffID` - this basically prevents having common code when integrating OCI save/load (#26369).
You can't get foreign sources _by diffID_ in OCI since all hashes are just from compressed layers digests and at the time we build this `manifestItem` for an OCI tarball we don't have info about diffIDs and cannot populate this map.

@aaronlehmann @stevvooe @tonistiigi any idea on how to convert this to something more useful to be used in the OCI PR also?